%YAML:1.0

# flags
use_imu: 0
use_lidar: 0
use_navsat: 1
num_of_cam: 2
is_semantic: 0

# ros parameters
imu_topic: '/kitti/oxts/imu'
lidar_topic: 'kitti/velodye_pointcloud'
navsat_topic: '/kitti/oxts/gps/fix'
image0_topic: '/kitti/camera_gray_left/image_raw'
image1_topic: '/kitti/camera_gray_right/image_raw'
result_path: '/home/jyp/Projects/lvio-fusion/result/result.csv'

# camera1 intrinsics
camera1.fx: 7.188560000000e+02
camera1.fy: 7.188560000000e+02
camera1.cx: 6.071928000000e+02
camera1.cy: 1.852157000000e+02

# camera2 intrinsics
camera2.fx: 7.188560000000e+02
camera2.fy: 7.188560000000e+02
camera2.cx: 6.071928000000e+02
camera2.cy: 1.852157000000e+02

# # body to camera1
# base_to_cam0: !!opencv-matrix
#    rows: 4
#    cols: 4
#    dt: d
#    data: [0.00874725,   -0.00479799,   0.99995,       1.10224,
#          -0.999864,     -0.0140061,    0.00867928,    -0.319072,
#          0.0139638,     -0.999891,     -0.00491985,   0.746066,
#          0,             0,             0,             1]

# # body to camera2
# base_to_cam1: !!opencv-matrix
#    rows: 4
#    cols: 4
#    dt: d
#    data: [-0.0226841,   -0.013533,     0.999651,      1.10288,
#          -0.999735,     0.00434584,    -0.0226272,    -0.856243,
#          -0.00403813,   -0.999899,     -0.013628,     0.749687,
#          0,             0,             0,             1]

# # body to lidar
# base_to_lidar: !!opencv-matrix
#    rows: 4
#    cols: 4
#    dt: d
#    data: [0.999998,     -0.000785403,  0.00202441,    0.810544,
#          0.000755307,   0.99989,       0.0148246,     -0.307054,
#          -0.00203583,   -0.014823,     0.999888,      0.802724,
#          0,             0,             0,             1]

# camera to base
base_to_cam0: !!opencv-matrix
   rows: 4
   cols: 4
   dt: d
   data: [0, 0, 1, 0,
          -1, 0, 0, 0,
          0, -1, 0, 0,
          0, 0, 0, 1]

# camera to lidar
base_to_cam1: !!opencv-matrix
   rows: 4
   cols: 4
   dt: d
   data: [0, 0, 1, 0,
          -1, 0, 0, -0.537165718864418,
          0, -1, 0, 0,
          0, 0, 0, 1]

# number of features
num_features: 200
num_features_init: 80
num_features_tracking: 120
num_features_tracking_bad: 8
num_features_needed_for_keyframe: 120

# backend
range: 5

